File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        minsoFar = prices[0]
        profit = 0

        for i in range(len(prices)):

            profit = max(profit, prices[i] - minsoFar)

            minsoFar = min(minsoFar, prices[i])


        return profit

        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int op = 0;
        int minInd = 0;
        int profit = 0;

        for(int i=0;i<prices.length;i++)
        {
            if(prices[i] < prices[minInd])
            {
                minInd = i;
            }

            profit = Math.max(profit, prices[i] - prices[minInd]);
        }

        return profit;
    }
}